corehq/apps/data_analytics/malt_generator.py

```diff
@@ -0,0 +1,75 @@
+from django.db import transaction
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.data_analytics.models import MALTRow
+from corehq.apps.domain.models import Domain
+from corehq.apps.smsforms.app import COMMCONNECT_DEVICE_ID
+from corehq.apps.sofabed.models import FormData
+
+
+class MALTTableGenerator(object):
+    """
+        Populates SQL table with data for given datespan
+        See MALTRow
+    """
+    _wam_pam_cache = {}
+
+    def __init__(self, datespan_object):
+        self.datespan = datespan_object
+
+    @transaction.atomic
+    def build_table(self):
+
+        for domain in Domain.get_all():
+            for user in domain.all_users():
+                forms_query = self.get_forms_queryset(user._id, domain.name)
```

Comment:
These three lines scare me.  There are about 10,000 domains and 100,000 users.  That means 10,000 database queries to get users per domain, then 100,000 iterations through everything else in this block:

1. a `query.count()` call
2. a normal filtered query
3. a db write per app, let's call that O(1)
4. a couch doc get

I understand this is an infrequently used task, so this might be an acceptable load as-is, I just want to make sure we think about it.

If that is problematically large, we should consider options for some/all of those db hits out of the loop.  It looks like there's a `MALTRow` created per user/app combination, but those could be aggregated in memory and bulk-saved.  Can either or both of the other two be moved out of the loop?  For instance, `1` would basically be "number of forms submitted per user (per domain)".  I don't know how to do that in SQL, but I believe it's possible.